### ISSUE TITLE
Consistently use "region" terminology in later stages of the compiler

### ISSUE THREAD

**Issue by WaffleLapkin** [rust-lang/compiler-team#634]
# Proposal

This includes the following renames:

- `GenericArgKind::{Lifetime => Region}`
- `tcx.{lifetimes => regions}`
    - `CommonLifetimes` => `CommonRegions`
    - `CommonRegions::{re_erased => erased, re_vars => vars, re_late_bounds => late_bounds}`
            - `re_static` was left as-is (`static` is a keyword...)

The reasoning is that most things in the later stages already use "region" terminology and this will be more consistent. As an example `GenericArgKind::Lifetime` currently wraps `ty::Region`, which is odd.

While we may want to consider different names in the future, I don't thinks that that conflicts with this proposal, if anything it'll be easier to rename to a different name if everything uses the same terminology; consistency in the moment matters.

# Mentors or Reviewers

This is a simple rename, that doesn't necessarily need mentors.

# Process

The main points of the [Major Change Process][MCP] are as follows:

* [x] File an issue describing the proposal.
* [ ] A compiler team member or contributor who is knowledgeable in the area can **second** by writing `@rustbot second`.
    * Finding a "second" suffices for internal changes. If however, you are proposing a new public-facing feature, such as a `-C flag`, then full team check-off is required.
    * Compiler team members can initiate a check-off via `@rfcbot fcp merge` on either the MCP or the PR.
* [ ] Once an MCP is seconded, the Final Comment Period begins. If no objections are raised after 10 days, the MCP is considered **approved**.

You can read [more about Major Change Proposals on forge][MCP].

[MCP]: https://forge.rust-lang.org/compiler/mcp.html

# Comments

**This issue is not meant to be used for technical discussion. There is a Zulip stream for that. Use this issue to leave procedural comments, such as volunteering to review, indicating that you second the proposal (or third, etc), or raising a concern that you would like to be addressed.**

**Comment by apiraino** [comment]
Closing MCP since supersed by #110254 (that proposal discussed on [Zulip](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/naming.20bikeshedding))

**Comment by WaffleLapkin** [comment]
@apiraino "since" is a bit weird, https://github.com/rust-lang/rust/issues/110254 was opened before this MCP...

**Comment by apiraino** [comment]
@WaffleLapkin you're right, I mixed up a bit the timeline of events. By reading the discussion [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/233931-t-compiler.2Fmajor-changes/topic/Consistently.20use.20.22region.22.20terminology.20in.20.E2.80.A6.20compiler-team.23634) I got that the agreement was that this proposal was going in the opposite direction of https://github.com/rust-lang/rust/issues/110254 (opened later tha this MCP) and https://github.com/rust-lang/compiler-team/issues/451 (started much earlier of this MCP AFAICS). Thus this proposal was not seconded.
